forest/cli/src/cli/snapshot_fetch.rs

```diff
@@ -102,22 +106,30 @@ async fn snapshot_fetch_mainnet(
     snapshot_out_dir: &Path,
     snapshot_fetch_config: &SnapshotFetchConfig,
 ) -> anyhow::Result<PathBuf> {
-    let client = Client::new();
-
-    let snapshot_url = snapshot_fetch_config.mainnet.snapshot_url.clone();
-    let snapshot_response = client.get(snapshot_url.clone()).send().await?;
-
-    // Use the redirect if available.
-    let snapshot_url = match snapshot_response
-        .headers()
-        .get("x-amz-website-redirect-location")
-    {
-        Some(url) => url.to_str()?.try_into()?,
-        None => snapshot_url,
```

Comment:
The old logic does not create new `snapshot_response` after redirection, it does seem logically correct although it works.

Reply:
The old code didn't use a HEAD request and only used the redirect location for naming the output file. Both approaches (using a HEAD request or not) are fine.